web/sharings/files.go

```diff
@@ -345,3 +418,72 @@ func createDirForSharing(fs vfs.VFS, id, name string) error {
 
 	return fs.CreateDir(dirDoc)
 }
+
+func mergeMetadata(newMeta, oldMeta vfs.Metadata) vfs.Metadata {
+	if newMeta == nil {
+		return oldMeta
+	}
+
+	res := vfs.Metadata{}
+	for newKey, newValue := range newMeta {
+		res[newKey] = newValue
+	}
+
+	for oldKey, oldValue := range oldMeta {
+		if _, present := newMeta[oldKey]; !present {
+			res[oldKey] = oldValue
+		}
+	}
+
+	return res
+}
+
+func mergeReferencedBy(newRefs, oldRefs []couchdb.DocReference) []couchdb.DocReference {
+	if len(newRefs) == 0 {
+		return oldRefs
+	}
+
+	res := make([]couchdb.DocReference, len(newRefs))
+	copy(res, newRefs)
+
+	for _, oldReference := range oldRefs {
+		var exists bool
+		for _, newReference := range newRefs {
+			if reflect.DeepEqual(newReference, oldReference) {
+				exists = true
+				break
+			}
+		}
+
+		if !exists {
+			res = append(res, oldReference)
+		}
+	}
```

Comment:
Well actually here I don't want to append if there are equal, the aim is to merge without duplicating elements (it does look like a mistake I made in a previous PR).